rueducommerce.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/chromedp/chromedp"
+	"time"
+)
+
+type Rueducommerce struct {
+	productUrl string
+	minPrice   float64
+}
+
+func (ru *Rueducommerce) Name() string {
+	return "RUEDUCOMMERCE"
+}
+
+func (ru *Rueducommerce) MinPrice() float64 {
+	return ru.minPrice
+}
+
+func (ru *Rueducommerce) FetchPrice(ctx context.Context) (float64, error) {
+	p, err := ru.getPrice(ctx)
+	if err != nil {
+		return InvalidPrice, fmt.Errorf("could not fetch price, got error %v", err)
+	}
+	return p, nil
+}
+
+func (ru *Rueducommerce) getPrice(ctx context.Context) (float64, error) {
+	var price float64
+	err := chromedp.Run(ctx,
+		chromedp.Navigate(ru.productUrl),
+		chromedp.Sleep(2*time.Second),
+		chromedp.Evaluate(ru.getPriceJS(), &price))
+	return price, err
+}
+
+func (ru *Rueducommerce) getPriceJS() string {
+	return fmt.Sprintf(`
+function convertPriceStringToFloat(price) {
+    return parseFloat(price.replace('€', '.'))
+}
+
+function getPrice() {
+    const priceElement = document.getElementsByClassName("price-pricesup")[0]
+    return priceElement && priceElement.innerText ? convertPriceStringToFloat(priceElement.innerText) : %d;
+}
+
+getPrice();
+`, InvalidPrice)
+}
```

Comment:
Easier to retrieve the price using:

```suggestion
func (ru *Rueducommerce) getPrice(ctx context.Context) (float64, error) {
	var price float64
	var ok bool
	err := chromedp.Run(ctx,
		chromedp.Navigate(ru.productUrl),
		chromedp.Sleep(2*time.Second),
		chromedp.AttributeValue("[itemprop='price']", "content", &price, &ok))
	if err != nil {
		return "", err
	}
	if !ok {
		return "", fmt.Errorf("failed to retrieve attribute value")
	}
	return price, nil
}
```